create

```diff
@@ -61,6 +61,17 @@ else
 fi
 
 mke2fs -Fjqt $OSP_FILESYSTEM $filesystem_dev
+
+# in some cases, mkfs will cause the partition tables to remap, for example
+# /dev/mapper/foo-1 will turn into /dev/mapper/foo-part1. Rebuild the map
+# to ensure the filesystem can be detected onward:
+if [ ! -b "$filesystem_dev" ]; then
```

Comment:
Minor bash style nit, it's better to use `[[ ]]` tests in bash.

Reply:
elsewhere the code doesn't use `[[ ]]`, so it would look odd. besides: why?

Reply:
There are some surprising gotchas with single brackets.

https://www.delftstack.com/howto/linux/difference-between-single-and-double-square-brackets-in-bash/